src/Features/Core/Portable/Testing/TestFrameworks/MSTestTestFrameworkMetadata.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Features.Testing;
+
+[Export(typeof(ITestFrameworkMetadata)), Shared]
+internal class MSTestTestFrameworkMetadata : ITestFrameworkMetadata
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public MSTestTestFrameworkMetadata()
+    {
+    }
+
+    public bool MatchesAttributeSymbolName(string attributeSymbolName)
+    {
+        return attributeSymbolName is "Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute";
```

Comment:
as before, i'd strongly recommend exposing the siple name (with or without the 'Attribute' suffix) so that a quick check can be done.  ideally a pure syntactic one IMO.